src/Carbon/Carbon.php

```diff
@@ -1984,6 +1984,58 @@ public function addYear($value = 1)
     }
 
     /**
+     * Add years to the instance with no overflow of months
```

Comment:
Can you base yourself on `\Carbon\Carbon::addMonths` so that you make use of `\Carbon\Carbon::shouldOverflowMonths` too ?